compiler/semstmts.nim

```diff
@@ -1519,20 +1588,17 @@ proc addResult(c: PContext, n: PNode, t: PType, owner: TSymKind) =
       n.add newSymNode(c.p.resultSym)
     addParamOrResult(c, c.p.resultSym, owner)
 
-proc copyExcept(n: PNode, i: int): PNode =
-  result = copyNode(n)
-  for j in 0..<n.len:
-    if j != i: result.add(n[j])
-
 proc semProcAnnotation(c: PContext, prc: PNode;
                        validPragmas: TSpecialWords): PNode =
   var n = prc[pragmasPos]
   if n == nil or n.kind == nkEmpty: return
   for i in 0..<n.len:
+    # Mirrored with semVarCustomPragma
+
     let it = n[i]
     let key = if it.kind in nkPragmaCallKinds and it.len >= 1: it[0] else: it
 
-    if whichPragma(it) != wInvalid:
+    if whichPragma(it) != wInvalid or key.kind notin nkIdentKinds:
```

Comment:
`considerQuotedIdent` fails on potentially valid pragmas without this check. A test that used `hint[XDeclaredButNotUsed]: off` on a template broke because of this.

Reply:
Don't comment here, write comments in the source code.